### PR TITLE
Resolve tenant domain at retrieving SCIM to local claim mapping.

### PR DESCRIPTION
**Pull request by Thisara-Welmilla**
When retrieving the SCIM to local claim mapping, the current implementation provide tenant domain of thread local service provider. The expected tenant domain is the user's tenant domain.

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java [hunk 1]
```diff
@@ -321,7 +321,7 @@ public static String formatDateTime(Date date) {
      */
     public static Map<String, String> getSCIMtoLocalMappings() throws UserStoreException {
 
-        String spTenantDomain = getTenantDomainFromSP();
+        String spTenantDomain = getTenantDomain();
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java:324]: can we do the same for customSchema as well and get rid of that  `String userTenantDomain = getTenantDomainFromContext();` and use this.

Also can we rename this spTenantDomain -> tenantDomain

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java [hunk 2]
```diff
@@ -359,22 +359,20 @@ public static String formatDateTime(Date date) {
...
-            return IdentityTenantUtil.getTenantDomainFromContext();
-        } else if (PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain() != null) {
-            tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            tenantDomain = IdentityTenantUtil.getTenantDomainFromContext();
         } else {
+            tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        }
+
+        if (tenantDomain.isEmpty()){
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java:375]: shall we use StringUtils.isBlank here

Comment (Thisara-Welmilla) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java]: Addressed with commit bb4f9622b8914f2c96b3c52ed8bb0b5373f6da3a

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java (hunk 1 above)
Comment (Thisara-Welmilla) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java]: Addressed with commit bb4f9622b8914f2c96b3c52ed8bb0b5373f6da3a

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java [hunk 3]
```diff
@@ -354,27 +354,25 @@ public static String formatDateTime(Date date) {
...
-            return IdentityTenantUtil.getTenantDomainFromContext();
-        } else if (PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain() != null) {
-            tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            tenantDomain = IdentityTenantUtil.getTenantDomainFromContext();
         } else {
+            tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        }
+
+        if (StringUtils.isBlank(tenantDomain)){
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java:375]: Shall we add some debug log here? Saying tenant domain is empty, and hence reading it as super tenant

Comment (Thisara-Welmilla) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java:375]: Addressed commit with 20ad7091a8daf313ae8ae76c4ddd8e74cda25b3c

**Review by piraveena — APPROVED** [review]

**Pull request merged**